drivers/gles2/rasterizer_scene_gles2.cpp

```diff
@@ -1385,21 +1385,241 @@ bool RasterizerSceneGLES2::_setup_material(RasterizerStorageGLES2::Material *p_m
+							if (s->format & VS::ARRAY_COMPRESS_NORMAL) {
+								const float multiplier = 1.f / 127.f;
+								const int8_t *v = (const int8_t *)(read.ptr() + offset);
+								attrib_array[0] = (float(v[0]) * multiplier) * base_weight;
+								attrib_array[1] = (float(v[1]) * multiplier) * base_weight;
+								attrib_array[2] = (float(v[2]) * multiplier) * base_weight;
+							} else {
+								const float *v = (const float *)(read.ptr() + offset);
+								attrib_array[0] = v[0] * base_weight;
+								attrib_array[1] = v[1] * base_weight;
+								attrib_array[2] = v[2] * base_weight;
+							}
+						} break;
+						case VS::ARRAY_TANGENT: {
+							if (s->format & VS::ARRAY_COMPRESS_TANGENT) {
+								const float multiplier = 1.f / 127.f;
+								const int8_t *v = (const int8_t *)(read.ptr() + offset);
+								attrib_array[0] = (float(v[0]) * multiplier) * base_weight;
+								attrib_array[1] = (float(v[1]) * multiplier) * base_weight;
+								attrib_array[2] = (float(v[2]) * multiplier) * base_weight;
+							} else {
+								const float *v = (const float *)(read.ptr() + offset);
+								attrib_array[0] = v[0] * base_weight;
+								attrib_array[1] = v[1] * base_weight;
+								attrib_array[2] = v[2] * base_weight;
+							}
+						} break;
+						case VS::ARRAY_COLOR: {
+							if (s->format & VS::ARRAY_COMPRESS_COLOR) {
+								const uint8_t *v = (const uint8_t *)(read.ptr() + offset);
+								attrib_array[0] = (v[0] / (float)0xFF) * base_weight;
+								attrib_array[1] = (v[1] / (float)0xFF) * base_weight;
+								attrib_array[2] = (v[2] / (float)0xFF) * base_weight;
+								attrib_array[3] = (v[3] / (float)0xFF) * base_weight;
+							} else {
+								const float *v = (const float *)(read.ptr() + offset);
+								attrib_array[0] = v[0] * base_weight;
+								attrib_array[1] = v[1] * base_weight;
+								attrib_array[2] = v[2] * base_weight;
+								attrib_array[3] = v[3] * base_weight;
+							}
+						} break;
+						case VS::ARRAY_TEX_UV: {
+							if (s->format & VS::ARRAY_COMPRESS_TEX_UV) {
+								const uint16_t *v = (const uint16_t *)(read.ptr() + offset);
+								attrib_array[0] = Math::halfptr_to_float(&v[0]) * base_weight;
+								attrib_array[1] = Math::halfptr_to_float(&v[1]) * base_weight;
+							} else {
+								const float *v = (const float *)(read.ptr() + offset);
+								attrib_array[0] = v[0] * base_weight;
+								attrib_array[1] = v[1] * base_weight;
+							}
+						} break;
+						case VS::ARRAY_TEX_UV2: {
+							if (s->format & VS::ARRAY_COMPRESS_TEX_UV2) {
+								const uint16_t *v = (const uint16_t *)(read.ptr() + offset);
+								attrib_array[0] = Math::halfptr_to_float(&v[0]) * base_weight;
+								attrib_array[1] = Math::halfptr_to_float(&v[1]) * base_weight;
+							} else {
+								const float *v = (const float *)(read.ptr() + offset);
+								attrib_array[0] = v[0] * base_weight;
+								attrib_array[1] = v[1] * base_weight;
+							}
+						} break;
```

Comment:
We will need a case for weights as well, it looks like they can just be blended the same as other attributes
https://github.com/godotengine/godot/blob/f34a8c7f0e9d127acc95ccff6839c7667ab736c0/drivers/gles3/shaders/blend_shape.glsl#L177-L181